modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,263 @@ class LinearAlgebraError : Error {
     }
 }
 
+pragma "no doc"
+/*
+This class promotes lazy computation of required
+Helper matrices to evaluate matrix exponential.
+
+High precision of expm requires smaller computations,
+hence we do a lazy computation depending on the norm value
+of the matrix.
+*/
+class ExpmPadeHelper {
+  type matType;
+  var A : matType;
+  var A2 : matType;
+  var A4 : matType;
+  var A6 : matType;
+  var A8 : matType;
+  var A10 : matType;
+  var Exactd4 : real;
+  var Exactd6 : real;
+  var Exactd8 : real;
+  var Exactd10 : real;
+  var Approxd4 : real;
+  var Approxd6 : real;
+  var Approxd8 : real;
+  var Approxd10 : real;
+  var Ident : matType;
+  var UseExactOneNorm : bool;
+  // Boolean arrays to know if A, Exactd, Approxd
+  // have been computed.
+  var isAComputed : ["A2", "A4", "A6", "A8", "A10"] bool;
+  var isExactdComputed : ["A4", "A6", "A8", "A10"] bool;
+  var isApproxdComputed : ["A4", "A6", "A8", "A10"] bool;
+
+  /*
+    Args:
+    A: Expects an N*N square matrix
+    UseExactOneNorm: boolean value specifying if the onenorm has to be exact.
```

Comment:
Unfortunately this still doesn't match the documentation style for arguments (though since the class itself is marked with `pragma "no doc"`, we won't see it in the documentation anyways).  It might be good practice to use the chpldoc syntax anyways, but I won't require it.